code/world.dm

```diff
@@ -108,414 +108,60 @@ var/global/datum/global_init/init = new ()
-			info["loc"] = M.loc ? "[M.loc]" : "null"
-			info["turf"] = MT ? "[MT] @ [MT.x], [MT.y], [MT.z]" : "null"
-			info["area"] = MT ? "[MT.loc]" : "null"
-			info["antag"] = M.mind ? (M.mind.special_role ? M.mind.special_role : "Not antag") : "No mind"
-			info["hasbeenrev"] = M.mind ? M.mind.has_been_rev : "No mind"
-			info["stat"] = M.stat
-			info["type"] = M.type
-			if(isliving(M))
-				var/mob/living/L = M
-				info["damage"] = list2params(list(
-							oxy = L.getOxyLoss(),
-							tox = L.getToxLoss(),
-							fire = L.getFireLoss(),
-							brute = L.getBruteLoss(),
-							clone = L.getCloneLoss(),
-							brain = L.getBrainLoss()
-						))
-			else
-				info["damage"] = "non-living"
-			info["gender"] = M.gender
-			return list2params(info)
-		else
-			return "Multiple matches found!"
-
-	else if(copytext(T,1,9) == "adminmsg")
-		/*
-			We got an adminmsg from IRC bot lets split the input then validate the input.
-			expected output:
-				1. adminmsg = ckey of person the message is to
-				2. msg = contents of message, parems2list requires
-				3. validatationkey = the key the bot has, it should match the gameservers commspassword in it's configuration.
-				4. sender = the ircnick that send the message.
-		*/
-
-
-		var/input[] = params2list(T)
-		var/bad_key = do_topic_spam_protection(addr, input["key"])
-
-		if (bad_key)
-			return bad_key
-
-		var/client/C
-		var/req_ckey = ckey(input["adminmsg"])
-
-		for(var/client/K in clients)
-			if(K.ckey == req_ckey)
-				C = K
-				break
-		if(!C)
-			return "No client with that name on server"
-
-		var/rank = input["rank"]
-		if(!rank)
-			rank = "Admin"
-
-		var/message =	"<font color='red'>Discord-[rank] PM from <b><a href='?discord_msg=[input["sender"]]'>Discord-[input["sender"]]</a></b>: [input["msg"]]</font>"
-		var/amessage =  "<font color='blue'>Discord-[rank] PM from <a href='?discord_msg=[input["sender"]]'>Discord-[input["sender"]]</a> to <b>[key_name(C)]</b> : [input["msg"]]</font>"
-
-		C.received_discord_pm = world.time
-		C.discord_admin = input["sender"]
-
-		C << 'sound/effects/adminhelp.ogg'
-		C << message
-
-
-		for(var/client/A in admins)
-			if(A != C)
-				A << amessage
-
-		return "Message Successful"
-
-	else if(copytext(T,1,6) == "notes")
-		var/input[] = params2list(T)
-		var/bad_key = do_topic_spam_protection(addr, input["key"])
-
-		if (bad_key)
-			return bad_key
-
-		return show_player_info_discord(ckey(input["notes"]))
-
-	else if(copytext(T,1,4) == "age")
-		var/input[] = params2list(T)
-		var/bad_key = do_topic_spam_protection(addr, input["key"])
-
-		if (bad_key)
-			return bad_key
-
-		var/age = get_player_age(input["age"])
-		if(isnum(age))
-			if(age >= 0)
-				return "[age]"
-			else
-				return "Ckey not found"
+	var/list/response[] = list()
+	var/list/queryparams[] = json_decode(T)
+	var/query = queryparams["query"]
+	var/auth = queryparams["auth"]
+	log_debug("API - Request Received - from:[addr], master:[master], key:[key]")

```

Comment:
These output to the same place, `log_debug` and `diary << "TOPIC: [...]"`

It might be an idea to replace them both with `log_misc("TOPIC: message goes here.")`. Misc logs are optional things that get shoved straight into the diary (day's log), if the config option is enabled.